server/game/cards/11-TCaR/DevilsSixGun.js

```diff
@@ -0,0 +1,66 @@
+const GoodsCard = require('../../goodscard.js');
+const GameActions = require('../../GameActions/index.js');
+
+class DevilsSixGun extends GoodsCard {
+    setupCardAbilities(ability) {
+        this.whileAttached({
+            effect: [
+                ability.effects.setAsStud(this.uuid),
+                ability.effects.modifyBullets(this.bullets)
+            ]
+        });
+        this.action({
+            title: 'Devil\'s Six Gun',
+            cost: ability.costs.bootSelf(),
+            playType: 'cheatin resolution',
+            handler: context => {
+                this.game.resolveGameAction(
+                    GameActions.search({
+                        title: 'Select a Joker',
+                        match: { condition: card => card.getType() === 'joker', location: ['draw deck', 'discard pile', 'dead pile'] },
+                        numToSelect: 1,
+                        message: {
+                            format: '{player} uses {source} to search for {searchTarget} and puts it into their draw hand'
+                        },
+                        cancelMessage: {
+                            format: '{player} uses {source} to search their draw deck, discard pile and Boot Hill, but does not find a Joker'
+                        },
+                        handler: jokerCard => {
+                            context.player.moveCard(jokerCard, 'draw hand');
+                            this.game.promptForSelect(context.player, {
```

Comment:
I have moved the discard select and event handling to the search handler. Thanks to this, we do not need global `selectedJoker` variable.

Reply:
As someone not familiar with this pull request tool...did you doing that erase my comment?

Reply:
yes, github automatically resolves review comments when the stuff you commented on is changed. it is not erased, just hidden/resolved.